OpenRA.Game/Map/Map.cs

```diff
@@ -317,7 +317,9 @@ public Map(TileSet tileset, int width, int height)
 		public Map(string path)
 		{
 			Path = path;
-			Container = Game.ModData.ModFiles.OpenPackage(path, null, int.MaxValue);
+			Container = Game.ModData.ModFiles.OpenPackage(path, null, int.MaxValue) as IReadWritePackage;
+			if (Container == null)
+				throw new InvalidDataException("Unsupported map package type");
```

Comment:
I would move this try-upcast-and-throw logic into a method such as `OpenPackageAsWritable` or such as I can imagine that would be a common pattern for anybody else who needs a writeable package.